### PR TITLE
provider/vsphere: Add insecure option for 'insecure' SSL requests

### LINKED ISSUE #3521: vsphere: turn on SSL verification and expose configuration option to disable
As @mkuzmin correctly pointed out in https://github.com/hashicorp/terraform/pull/3419/files#r42080780, the default here is insecure.

Good precent here is an `allow_unverified_ssl` parameter from rundeck:

https://github.com/hashicorp/terraform/blob/master/builtin/providers/rundeck/provider.go#L25-L30

### PR DESCRIPTION
**Pull request by tkak**
This fixes #3521.

### CODE REVIEW

File: builtin/providers/vsphere/provider.go [hunk 1]
```diff
@@ -29,6 +29,13 @@ func Provider() terraform.ResourceProvider {
 				DefaultFunc: schema.EnvDefaultFunc("VSPHERE_VCENTER", nil),
 				Description: "The vCenter Server name for vSphere API operations.",
 			},
+
+			"insecure": &schema.Schema{
```
Comment (jen20) [inline builtin/providers/vsphere/provider.go]: I think we should consider naming this `allow_unverified_ssl` such that it consistent with the [Rundeck provider](https://github.com/hashicorp/terraform/blob/master/builtin/providers/rundeck/provider.go#L25-L30), and in order that it describes more clearly exactly which form of insecurity is invited?

**Comment by jen20** [comment]
Hi @tkak! Thanks for opening up this PR. I've left some comments inline; could you review them and let me know your thoughts?

File: builtin/providers/vsphere/provider.go (hunk 1 above)
Comment (phinze) [inline builtin/providers/vsphere/provider.go]: +1 lets use `allow_unverified_ssl` as the parameter name here.

**Comment by chrislovecnm** [comment]
@tkak you have a couple of moments to tweak the name??

**Comment by tkak** [comment]
@jen20 Thanks for the review! For the reason of `insecure`, I referenced the usage of VMware vSphere go library.
https://github.com/vmware/govmomi/blob/8f2d79fd740b69c4f64d10e8ac0b8c6b30247c0d/examples/datastores/main.go#L70.

And `insecure` parameter is used at the [OpenStack provider](https://terraform.io/docs/providers/openstack/index.html#insecure).

**Comment by jen20** [comment]
Hi @tkak - In this case I still think `allow_unverified_ssl` is better than `insecure` (and that we should actually update the OpenStack provider as well and deprecate `insecure`).

**Comment by jtopjian** [comment]
(I have gmail filters for "openstack" so I saw this conversation :smile: )

I'm mixed on the rename. On one hand, I totally agree that `allow_unverified_ssl` is more intuitive than `insecure`, but on the other hand, `insecure` is used widely outside of Terraform.

**Comment by chrislovecnm** [comment]
@jen20 @jtopjian @phinze - who can make a call?  I would like to just get this in :laughing:

**Comment by tkak** [comment]
@jen20 I've fixed `insecure` parameter to `allow_unverified_ssl` :smile: . Because I think intuitiveness is more important than precedents for TF configuration file.

**Comment by phinze** [comment]
This looks good - just needs a rebase and we can get it merged. :+1:

**Comment by phinze** [comment]
I too prefer the more intuitive name over respecting precedent, provided we are doing what we can to remain consistent within Terraform.

**Comment by tkak** [comment]
@phinze I tried to rebase. :+1:

**Pull request closed without merging**